### PR TITLE
Allowing failures on acceptance tests due to bug in Gitlab

### PR DESCRIPTION
**Pull request by brandonrdn**

### CODE REVIEW

**Comment by brandonrdn** [comment]
CI job: https://gitlab.com/simp/pupmod-simp-simp_gitlab/pipelines/25216014

**Comment by jeefberkey** [comment]
:heavy_check_mark: https://gitlab.com/simp/pupmod-simp-simp_gitlab/pipelines/25223539

File: spec/acceptance/suites/default/10_ldaps_spec.rb [hunk 1]
```diff
@@ -156,12 +156,15 @@ class { 'simp_gitlab':
...
-        expect(noko_alert_text).to_not match(/^Could not authenticate/)
-
-        # Test for success
-        expect(profile_link).not_to be_empty
-        expect(profile_link.first['data-user']).to eq('ldapuser1')
+      ## Allowing failures due to a bug in Gitlab, referenced in ticket SIMP-4946:
+      ## https://simp-project.atlassian.net/browse/SIMP-4946
+      ## Uncomment this after A) We work around the issue, or B) Gitlab resolves the issue in 11.1
+      ## # Test for failure
+      ##   expect(noko_alert_text).to_not match(/^Could not authenticate/)
```
[the remark this answers is not in the document]
Comment (op-ct) [reply spec/acceptance/suites/default/10_ldaps_spec.rb]: BTW, although I mentioned using `:skip`, you can also use `:pending`.

Tests that are `:pending` will still run, but are _expected_ to fail—and if they don't, RSpec reports them as failed expectations and flags them as `FIXED`:

```
Failures:

   1) gitlab ssh access user ssh GitLab access FIXED
      Expected pending 'should allow user to clone via ssh' to fail. No error was raised.
      # ./spec/acceptance/suites/default/11_ssh_access_spec.rb:100
```

The tradeoff between the two is that—although `:pending` can tip you off that an upstream problem (like this GitLab/omniauth-ldap bug) has been fixed—it still runs the tests.  But if those tests take a long time, or use expensive resources, or spam the test report logs with hideous garbage, you can use `:skip` and just keep a closer eye on the upstream situation so you know when to re-enable things.

**Comment by brandonrdn** [comment]
@op-ct Done. Updated CI pipeline: https://gitlab.com/simp/pupmod-simp-simp_gitlab/pipelines/25230185

**Comment by brandonrdn** [comment]
@trevor-vaughan fixed whitespace. I like the idea of adding the check for gitlab. Maybe in a minor release between the beta and release?

**Comment by trevor-vaughan** [comment]
@brandonrdn If there is time, otherwise, it can happen whenever since it's not a critical part of the actual release.

**Pull request merged**